### PR TITLE
Use string interpolation where possible

### PR DESCRIPTION
**Pull request by ZivDero**
This PR replaces all instances of string concatenation (and string.Format) with string interpolation.
- String interpolation is marginally more efficient as it uses StringBuilder
- Interpolated strings are easier to read

### CODE REVIEW

File: src/TSMapEditor/CCEngine/VxlFile.cs [hunk 1]
```diff
@@ -44,11 +44,12 @@ public VxlFile(byte[] buffer, string filename = "") : base(new MemoryStream(buff
...
         {
             if (Length < FileHeader.Size)
-                throw new VxlLoadException(nameof(VxlFile) + " .vxl is shorter than specified in its header!");
+                throw new VxlLoadException($"{nameof(VxlFile)} .vxl is shorter than specified in its header!");
 
             Header.Read(this);
             if (Header.HeaderCount == 0 || Header.TailerCount == 0 || Header.TailerCount != Header.HeaderCount)
-                throw new VxlLoadException(nameof(VxlFile) + " .vxl has no header or tailer sections, or their count doesn't match!");
+                throw new VxlLoadException(
+                    $"{nameof(VxlFile)} .vxl has no header or tailer sections, or their count doesn't match!");
```
Comment (Rampastring) [inline src/TSMapEditor/CCEngine/VxlFile.cs:52]: Splitting this into another line seems pointless. It also breaks our contribution guidelines.

**Comment by ZivDero** [comment]
Looking back, I will go over this in detail and restore concatenation in some places where it makes it more readable (mostly very long lines with breaks).

File: src/TSMapEditor/CCEngine/ShpFile.cs [hunk 2]
```diff
@@ -254,8 +255,8 @@ private void DecompressRLEZero(byte[] frameData, int frameIndex, ShpFrameInfo fr
                     {
                         if (pixelPositionOnLine >= frameInfo.Width)
                         {
-                            throw new ShpLoadException($"Out-of-bounds pixel position on color data in SHP RLE-Zero frame. " +
-                                $"File name: {fileName}, frame index: {frameIndex}, line index: {lineIndex}, file offset: {lineDataStartOffset + currentByteIndex}");
+                            throw new ShpLoadException(
+                                $"Out-of-bounds pixel position on color data in SHP RLE-Zero frame. File name: {fileName}, frame index: {frameIndex}, line index: {lineIndex}, file offset: {lineDataStartOffset + currentByteIndex}");
```
Comment (Rampastring) [inline src/TSMapEditor/CCEngine/ShpFile.cs:259]: I think the original formatting here was superior with it not having such a long line.

File: src/TSMapEditor/Initialization/MapLoader.cs [hunk 3]
```diff
@@ -84,8 +84,8 @@ public static void PostCheckMap(IMap map, TheaterGraphics theaterGraphics)
                 int maxSubTileIndex = tile.SubTileCount - 1;
                 if (t.SubTileIndex > maxSubTileIndex)
                 {
-                    AddMapLoadError($"Invalid sub-tile index {t.SubTileIndex} for cell at {t.CoordsToPoint()} (max: {maxSubTileIndex}) - setting it to 0. " +
-                        $"TileSet: {tileSet.SetName} ({tileSet.FileName}), index of tile within its set: {tile.TileIndexInTileSet}");
+                    AddMapLoadError(
+                        $"Invalid sub-tile index {t.SubTileIndex} for cell at {t.CoordsToPoint()} (max: {maxSubTileIndex}) - setting it to 0. TileSet: {tileSet.SetName} ({tileSet.FileName}), index of tile within its set: {tile.TileIndexInTileSet}");
```
Comment (Rampastring) [inline src/TSMapEditor/Initialization/MapLoader.cs:88]: I think the original formatting here was superior with it not having such a long line.

File: src/TSMapEditor/Initialization/MapWriter.cs [hunk 4]
```diff
@@ -599,17 +586,8 @@ public static void WriteBuildings(IMap map, IniFile mapIni)
...
-                               $"{BoolToObjectStyle(structure.AISellable)}," +
-                               $"{BoolToObjectStyle(structure.AIRebuildable)}," +
-                               $"{BoolToObjectStyle(structure.Powered)}," +
-                               $"{structure.UpgradeCount}," +
-                               $"{(int)structure.Spotlight}," + 
-                               $"{upgrade1},{upgrade2},{upgrade3}," +
-                               $"{BoolToObjectStyle(structure.AIRepairable)}," +
-                               $"{BoolToObjectStyle(structure.Nominal)}";
+                string value =
+                    $"{structure.Owner.ININame},{structure.ObjectType.ININame},{structure.HP},{structure.Position.X},{structure.Position.Y},{structure.Facing},{attachedTag},{BoolToObjectStyle(structure.AISellable)},{BoolToObjectStyle(structure.AIRebuildable)},{BoolToObjectStyle(structure.Powered)},{structure.UpgradeCount},{(int)structure.Spotlight},{upgrade1},{upgrade2},{upgrade3},{BoolToObjectStyle(structure.AIRepairable)},{BoolToObjectStyle(structure.Nominal)}";
```
Comment (Rampastring) [inline src/TSMapEditor/Initialization/MapWriter.cs:590]: Ew.

File: src/TSMapEditor/Rendering/GameClass.cs [hunk 5]
```diff
@@ -50,29 +51,28 @@ public GameClass()
...
             }
 
             Logger.Log("Exiting.");
 
             windowManager?.HideWindow();
-            System.Windows.Forms.MessageBox.Show("The map editor has crashed." + Environment.NewLine + Environment.NewLine +
-                "Exception information logged into except.txt:" + Environment.NewLine + Environment.NewLine +
-                sb.ToString());
+            System.Windows.Forms.MessageBox.Show(
+                $"The map editor has crashed.{Environment.NewLine}{Environment.NewLine}Exception information logged into except.txt:{Environment.NewLine}{Environment.NewLine}{sb}");
```
Comment (Rampastring) [inline src/TSMapEditor/Rendering/GameClass.cs:75]: The original formatting here nicely split the lines just like they are split in the resulting message box.

**Review by Rampastring — COMMENTED** [review]
I actually disagree about interpolated strings being easier to read in all cases. For example

```cs
throw new FileNotFoundException("Primary MIX file not found: " + name);
throw new FileNotFoundException($"Primary MIX file not found: {name}");
```

In cases like this there's no clear difference. I actually find the non-interpolated slightly easier to read, although likely only because I'm used to it.

Where interpolation is clearly superior is when there's multiple arguments to the string. Such as this case

```cs
$"Failed to load CSF file. Make sure that the file is not corrupted. Filename: {fileName}, original exception: {ex.Message}"
```

This means that interpolation itself can be used in all cases without any real drawbacks. Which means that I could approve the core idea of this PR.

However, outside of interpolation itself, this PR has some very ugly newline changes, probably as a result of doing this change with some tool that uses "generic" code formatting rules without caring about the context. I highlighted a few, but there are far more in the code.

This is why I'm generally careful with code refactoring tools and don't like code style enforcers, as they tend to always do something worse compared to my own, manual work.

**Pull request closed without merging**